### PR TITLE
[Feature][Connector-V2] Add Dingtalk Sink #2257

### PR DESCRIPTION
**Pull request by MRYOG**
Add Dingtalk Sink #2257 ,include sink and api example

### CODE REVIEW

**Comment by ic4y** [comment]
cc @Hisoka-X

File: pom.xml [hunk 1]
```diff
@@ -87,7 +87,7 @@
         <module>seatunnel-plugin-discovery</module>
         <module>seatunnel-formats</module>
         <module>seatunnel-dist</module>
-	    <module>seatunnel-server</module>
+	      <module>seatunnel-server</module>
```
Comment (Hisoka-X) [inline pom.xml]:
```suggestion
	  <module>seatunnel-server</module>
```

File: seatunnel-connectors-v2/connector-dingtalk/pom.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>seatunnel-connectors-v2</artifactId>
+    <groupId>org.apache.seatunnel</groupId>
+    <version>2.1.3-SNAPSHOT</version>
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-dingtalk/pom.xml]:
```suggestion
    <version>${revision}</version>
```

File: seatunnel-connectors-v2/connector-dingtalk/pom.xml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>connector-dingtalk</artifactId>
+
+  <properties>
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-dingtalk/pom.xml]: Please remove maven jdk version. We should make sure our connector can compiled success both on jdk8 and jdk11.

File: seatunnel-connectors-v2/connector-dingtalk/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DingTalkSink.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  private Config pluginConfig;
+  private SeaTunnelRowType seaTunnelRowType;
+  @Override
+  public String getPluginName() {
+    return "DingTalk";
+  }
+
+  @Override
+  public void prepare(Config pluginConfig) throws PrepareFailException {
+    this.pluginConfig = pluginConfig;
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-dingtalk/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DingTalkSink.java]: Please check your config, make sure `url` and `secret` parameter must included in pluginConfig.

File: seatunnel-examples/seatunnel-flink-connector-v2-example/src/main/java/org/apache/seatunnel/example/flink/v2/SeaTunnelApiExample.java [hunk 5]
```diff
@@ -31,7 +31,7 @@
 public class SeaTunnelApiExample {
 
     public static void main(String[] args) throws FileNotFoundException, URISyntaxException, CommandException {
-        String configFile = getTestConfigFile("/examples/fake_to_console.conf");
+        String configFile = getTestConfigFile("/examples/fake_to_dingtalk.conf");
```
Comment (Hisoka-X) [inline seatunnel-examples/seatunnel-flink-connector-v2-example/src/main/java/org/apache/seatunnel/example/flink/v2/SeaTunnelApiExample.java]: Please don't change this default config. We should make sure other developer can run this example easy. You can create a SeaTunnelDingTaskExample.java or create a e2e test for DingTalk.

**Comment by Hisoka-X** [comment]
Also please add DingTalk properties to plugin-mapping.properties file.

**Comment by MRYOG** [comment]
Got it

**Review by EricJoy2048 — CHANGES_REQUESTED** [review]
Please add the document for this connector. The path is `docs/en/connector-v2/sink`.
Thanks.

**Comment by MRYOG** [comment]
> Please add the document for this connector. The path is `docs/en/connector-v2/sink`. Thanks.

**Review by EricJoy2048 — COMMENTED** [review]
Please don't update this file `.idea/vcs.xml`

**Comment by EricJoy2048** [comment]
Please add `connector-dingtalk` to `seatunnel-connector-v2-dist/pom.xml`.

**Comment by Hisoka-X** [comment]
Hi Please resolve ci problem. Thanks!

File: pom.xml (hunk 1 above)
Comment (MRYOG) [reply pom.xml]: done

File: seatunnel-connectors-v2/connector-dingtalk/pom.xml (hunk 2 above)
Comment (MRYOG) [reply seatunnel-connectors-v2/connector-dingtalk/pom.xml]: done

File: seatunnel-connectors-v2/connector-dingtalk/pom.xml (hunk 3 above)
Comment (MRYOG) [reply seatunnel-connectors-v2/connector-dingtalk/pom.xml]: done

File: seatunnel-connectors-v2/connector-dingtalk/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DingTalkSink.java (hunk 4 above)
Comment (MRYOG) [reply seatunnel-connectors-v2/connector-dingtalk/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DingTalkSink.java]: done

File: seatunnel-examples/seatunnel-flink-connector-v2-example/src/main/java/org/apache/seatunnel/example/flink/v2/SeaTunnelApiExample.java (hunk 5 above)
Comment (MRYOG) [reply seatunnel-examples/seatunnel-flink-connector-v2-example/src/main/java/org/apache/seatunnel/example/flink/v2/SeaTunnelApiExample.java]: done

**Comment by MRYOG** [comment]
add doc , dingtalk.md
add v2-dist  pom.xml
add  plugin-mapping.properties
add DingTalk example
fix {reversion}
confim vcs.xml , pom.xml

**Comment by MRYOG** [comment]
> add doc , dingtalk.md add v2-dist pom.xml add plugin-mapping.properties add DingTalk example fix {reversion} confim vcs.xml , pom.xml

It's done

**Comment by Hisoka-X** [comment]
Please fix ci error.

**Comment by MRYOG** [comment]
Sorry, my checkstyle is not seatunnel config, so it has many problem.

**Comment by EricJoy2048** [comment]
> Sorry, my checkstyle is not seatunnel config, so it has many problem.

SeaTunnel checkstyle file in `tools/checkstyle`

**Comment by MRYOG** [comment]
> > Sorry, my checkstyle is not seatunnel config, so it has many problem.
> 
> SeaTunnel checkstyle file in `tools/checkstyle`

Yes,I fix it

**Comment by EricJoy2048** [comment]
cc @CalvinKirs  @Hisoka-X  First-time contributors need a maintainer to approve running workflow

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**